private-templates-service/client/src/App.tsx

```diff
@@ -72,39 +81,63 @@ interface Props {
   getOrgDetails: () => void;
   getProfilePicture: () => void;
   userLogout: () => void;
+  getConfig: () => void;
   isAuthenticated: boolean;
   user?: UserType;
   searchByTemplateName: string;
+  appId?: string;
+  redirectUri?: string;
+  appInsightsInstrumentationKey?: string;
 }
 
 class App extends Component<Props, State> {
-  userAgentApplication: UserAgentApplication;
+  userAgentApplication?: UserAgentApplication;
   onIdle: () => any;
 
-  constructor(props: Props) {
-    super(props);
-    initializeIcons();
-    this.userAgentApplication = new UserAgentApplication({
-      auth: {
-        clientId: process.env.REACT_APP_ACMS_APP_ID!,
-        redirectUri: process.env.REACT_APP_ACMS_REDIRECT_URI
-      },
-      cache: {
-        cacheLocation: "localStorage",
-        storeAuthStateInCookie: true
+  componentDidUpdate() {
+    if (!this.userAgentApplication && this.props.appId && this.props.redirectUri) {
+      this.userAgentApplication = new UserAgentApplication({
+        auth: {
+          clientId: this.props.appId!,
+          redirectUri: this.props.redirectUri
+        },
+        cache: {
+          cacheLocation: "localStorage",
+          storeAuthStateInCookie: true
+        }
+      });
+  
+      let user = this.userAgentApplication.getAccount();
+  
+      if (user) {
+        // Enhance user object with data from Graph
+        this.getUserInfo();
       }
-    });
 
-    let user = this.userAgentApplication.getAccount();
+      if (this.props.appInsightsInstrumentationKey) {
+        const browserHistory = createBrowserHistory({ basename: process.env.REACT_APP_ACMS_APP_NAME });
+        var reactPlugin = new ReactPlugin();
+        var appInsights = new ApplicationInsights({
+          config: {
+            instrumentationKey: process.env.REACT_APP_ACMS_APP_INSIGHTS_INSTRUMENTATION_KEY,
```

Comment:
If the instrumentation key isn't set, i think it fail. Can we add something like this: 
`instrumentationKey: process.env.REACT_APP_ACMS_APP_INSIGHTS_INSTRUMENTATION_KEY || "test"`

Reply:
I think it'll be okay, if the environment variable for the instrumentation key is not set, the backend will not return it at /config and the value will be undefined, so this step will be skipped

Reply:
even better!